database/engine/datafile.c

```diff
@@ -157,7 +274,7 @@ int create_data_file(struct rrdengine_datafile *datafile)
 
     ret = posix_memalign((void *)&superblock, RRDFILE_ALIGNMENT, sizeof(*superblock));
     if (unlikely(ret)) {
-        fatal("posix_memalign:%s", strerror(ret));
+        fatal("DBENGINE: posix_memalign:%s", strerror(ret));
```

Comment:
should it be `superblock->tier = ctx->tier + 1;` few lines below instead of fixed `    superblock->tier = 1;` 

same goes for `void generate_datafilepath(struct rrdengine_datafile *datafile, char *str, size_t maxlen)`

it seems to always output 1 as tier as it uses `datafile->tier` instead of `datafile->ctx->tier`?

Reply:
The tier in file name is not related to the ctx->tier. The file name convention was "reserved" for tier implementation (it would play well with filename sorting when deciding which files to delete) but we end up was implemented differently.